### PR TITLE
Buy me a coffee tutorial

### PR DESCRIPTION
**Pull request by ayo-klaytn**
## Proposed changes

Buy me a coffee tutorial was added

## Types of changes

Please put an x in the boxes related to your change.

- [x] Minor Issues and Typos
- [ ] Major Content Contribution
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to reach out. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn-docs/blob/master/CONTRIBUTING.md)
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn-docs)
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large content contribution, kick off the discussion by explaining why you would suggest the content contribution, etc...

### CODE REVIEW

File: docs/dapp/tutorials/buy-me-a-coffee/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+* [1. Project Setup](1.-project-setup.md)
+* [2. Creating a Buy Me A Coffee Smart Contract](2.-creating-bmc-contract.md)
+* [3. Testing the contract’s functionalities using scripts](3.-testing-bmc-contract-using-scripts.md)
+* [4. Deploying BMC Smart contract to Klaytn Testnet ](4.-deploying-bmc-contract.md)
+* [5. Building the BMC Frontend with React and Web3Onboard](5.-bmc-frontend-with-react-and-web3onboard.md)
+* [6. Deploying Frontend code on IPFS using Fleek](6.-deploying-frontend-to-fleek.md)
+* [7. Conclusion](7.-conclusion.md)
+
+## 1. Introduction <a href="#1-introduction" id="1-introduction"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/README.md:13]: "1. Introduction" to "Introduction"

File: docs/dapp/tutorials/buy-me-a-coffee/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+In this tutorial. you will build a decentralized version of the Buy Me a Coffee (BMC) platform (frontend + smart contract). This platform will be a minimalistic implementation of the traditional BMC platform where supporters can tip you, and you will be able to withdraw any tips that are delivered to the BMC smart contract as the contract's owner. Supporters will be able to send test KLAY and lovely messages together in a coffee transaction using this site. 
+
+By the end of this guide, you will have used the following to create this dApp: 
+- Solidity: to write the BMC smart contract
+- NextJs and Tailwind: for building a frontend website for our BMC dApp
+- Web3Onboard: to enable multiple wallet connections to Klaytn Testnet Baobab.
+- Fleek: with Fleek we can host our BMC dApp on IPFS.
+
+## 2. Prerequisites <a href="#2-prerequisites" id="2-prerequisites"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/README.md:31]: "2. Prerequisites" to "Prerequisites"

File: docs/dapp/tutorials/buy-me-a-coffee/2.-creating-bmc-contract.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Creating a Buy Me A Coffee Smart Contract <a id="creating-a-buy-me-a-coffee-contract"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/2.-creating-bmc-contract.md:1]: "Creating a ... " to "2. Creating a ..."

File: docs/dapp/tutorials/buy-me-a-coffee/3.-testing-bmc-contract-using-scripts.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
+# Testing the contract’s functionalities using scripts <a id="testing-bmc-contract-using-scripts"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/3.-testing-bmc-contract-using-scripts.md:1]: "Testing ..." to "3. Testing ..."

File: docs/dapp/tutorials/buy-me-a-coffee/4.-deploying-bmc-contract.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
+# 1. Deploying BMC Smart contract to Klaytn Testnet  <a id="deploying-bmc-contract"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/4.-deploying-bmc-contract.md:1]: "1. Deploying ..." to "4. Deploying ..."

File: docs/dapp/tutorials/buy-me-a-coffee/4.-deploying-bmc-contract.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+npx hardhat run scripts/deploy.js --network baobab
+```
+Once the contract deploys successfully, your terminal should look like this:
+
+```bash
+BuyMeACoffee Contract Address 0x0bEd1ed7B205d8c18e38A20b5BaB6e265A96d1AC
+```
+Congratulations on deploying your BMC smart contract on Klaytn Baobab Network! You can verify this transaction on Klaytnscope  by pasting your address in the search field.
+
+# Interacting with BMC Smart Contract  <a id="interacting-with-bmc-contract"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/4.-deploying-bmc-contract.md:72]: Are you sure this should be the heading level 1? (Just out of curiosity)

File: docs/dapp/tutorials/buy-me-a-coffee/5.-bmc-frontend-with-react-and-web3onboard.md [hunk 7]
```diff
@@ -0,0 +1,284 @@
+# Building the BMC Frontend with React and Web3Onboard <a id="builidng-bmc-frontend-with-react-and-web3onboard"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/5.-bmc-frontend-with-react-and-web3onboard.md:1]: "Building ..." to "5. Building ..."

File: docs/dapp/tutorials/buy-me-a-coffee/5.-bmc-frontend-with-react-and-web3onboard.md [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+The next step is to install the necessary dependencies to get our BMC frontend website up and running.  The following are the packages to be installed: 
+
+1. Web3Onboard packages: Web3-Onboard is a chain-agnostic wallet library that supports multi-wallet compatibility in your dApp built on EVM-compatible networks like Klaytn Blockchain.
+2. ethers.js: Web3-Onboard provider can be used with libraries like [ethers.js](https://docs.ethers.org/v6/) and[web3.js](https://web3js.readthedocs.io/en/v1.2.8/getting-started.html). In this guide, we will use ethers.js to make Klaytn blockchain calls like getting the user's account, fetch balance, sign transaction, send transaction, read from and write to the smart contract.
+
+Important Note: We need to edit 2 files in the  frontend/pages folder
+- **_app.js**
+- **index.js**
+
+## Setting up Web3Onboard Provider and Wallet Modules <a id="setting-up-web3onboard-provider-and-wallet-modules"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/5.-bmc-frontend-with-react-and-web3onboard.md:18]: "Setting up" to "5.1 Setting up"

File: docs/dapp/tutorials/buy-me-a-coffee/6.-deploying-frontend-to-fleek.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
+# Deploying Frontend code on IPFS using Fleek <a id="deploying-bmc-frontend-to-ipfs-using-fleek"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/6.-deploying-frontend-to-fleek.md:1]: "Deploying ... " to "6. Deploying "

File: docs/dapp/tutorials/buy-me-a-coffee/7.-conclusion.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Conclusion <a id="conclusion"></a>
```
Comment (Scott-Klatn) [inline docs/dapp/tutorials/buy-me-a-coffee/7.-conclusion.md:1]: "Conclusion" to "7. Conclusion"

**Review by Scott-Klatn — COMMENTED** [review]
I left some comments with regard to consistency in numbering of headings. Basically it's not common and advisable to prefix numbers to headings or sub headings manually in technical documentation, because it's quite likely that we will need to update them one by one if the relevant information is updated or rearranged. I, however, noticed there already have been the cases in the previous content, so I just want to make it consistent at least.

File: docs/dapp/tutorials/buy-me-a-coffee/4.-deploying-bmc-contract.md (hunk 6 above)
Comment (ayo-klaytn) [reply docs/dapp/tutorials/buy-me-a-coffee/4.-deploying-bmc-contract.md:72]: Yea...probably a 4.2 heading and deploying  bmc contract 4.1

**Review by praveen-klaytn — APPROVED** [review]

**Review by Scott-Klatn — APPROVED** [review]

**Pull request merged**